modules/scheduler/gke-cluster/main.tf

```diff
@@ -315,3 +317,52 @@ module "workload_identity" {
     google_container_cluster.gke_cluster
   ]
 }
+
+data "google_client_config" "default" {}
+
+provider "kubectl" {
+  host                   = "https://${google_container_cluster.gke_cluster.endpoint}"
+  cluster_ca_certificate = base64decode(google_container_cluster.gke_cluster.master_auth[0].cluster_ca_certificate)
+  token                  = data.google_client_config.default.access_token
+  load_config_file       = false
+}
+
+resource "kubectl_manifest" "additional_net_params" {
+  for_each = { for idx, network_info in var.additional_networks : idx => network_info }
+
+  depends_on = [google_container_cluster.gke_cluster]
+
+  yaml_body = <<YAML
+apiVersion: networking.gke.io/v1
+kind: GKENetworkParamSet
+metadata:
+  name: vpc${each.key + 1}
+spec:
+  vpc: ${each.value.network}
+  vpcSubnet: ${each.value.subnetwork}
+  deviceMode: NetDevice
+YAML
+
+  provider = kubectl
+}
+
+resource "kubectl_manifest" "additional_nets" {
```

Comment:
Is this manifest always applied ? Or only when the machine type is a3-mega or a3-highgpu ?

Reply:
Whenever a multivpc is passed to the module.

Reply:
ack, now I see the for loop with additional_networks , thanks!